packages/http/src/forwarder/HttpForwarder.ts

```diff
@@ -18,7 +19,7 @@ export class HttpForwarder implements IForwarder<IHttpOperation, IHttpRequest, I
         : inputData.url.baseUrl;
 
     if (!baseUrl) {
-      throw new Error('Either one server in spec or baseUrl in request must be defined');
+      throw ProblemJsonError.fromTemplate(NO_BASE_URL_ERROR);
```

Comment:
Just FYI, I always have a problem with this. It is so very surprising to see a `return` at the end of the function and then, later, learn that we might never get to this `return`. I'd personally change this `if` to `if/else`. In `if` we throw, in `else`, we `return`. But I'm aware that some use this `early return` practice. I don't agree with it, it's so confusing.

Reply:
Oh yes it's been debated many times and I alone have participated in such debates many times :) 
To else or not to else that is the question!

Now after all the years writing JS code I think both styles are correct and I personally let people choose their liking.
I think in general these days styling issues should be picked up by linter to be quite honest.
I remember times when 90% of code reviews would revolve around code style and in the end no one could really win the argument leaving less time to focus on functional or design flaws.

I personally like to reduce the nesting, therefore I don't add the 'else' statement if it's not logically required.

Reply:
But it's not about styling. Let's see:
`if x then y else z`
```
if x then y
z
```
The first example clearly shows logical flow (how it is branching out). In the second one, there is nothing that shows this connection. As I said, I prefer the first one and deem it clearer, but won't argue here since this is so old and unsettled issue.

Reply:
Ha! I always love these discussions where everybody is right and nobody is wrong.

I won't refrain from adding my (useless) two cent:

Historically the early return was used because some compilers could effectively generate branchless code for it:

```ts
if (a) 
  return 0;
return 1;
```

Some compilers would effectively rewrite the output to be more like this:

```ts
const results = [0, 1];
return results[a];
```

Which effectively is a branchless code.

Today this is probably not true anymore — we're not writing code that's compiled, we're writing suggestions to the interpreter. Therefore clarity *can* be preferred.